### PR TITLE
TA 90146

### PR DESCRIPTION
**Pull request by gemma-shay**
https://cockroachlabs.atlassian.net/browse/DOC-5988

### CODE REVIEW

**Review by shermanCRL — APPROVED** [review]
Thank you!

File: advisories/a90146.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: Technical Advisory 90146
+advisory: A-90146
+summary: Changefeeds using the <code>initial_scan='only'</code> or <code>schema_change_policy='stop'</code> options may incorrectly complete with a <code>successful</code> status under certain circumstances.
```
Comment (taroface) [inline advisories/a90146.md:4]: I don't think we need the `code` tags in the summary, unless you've seen this successfully used elsewhere.

File: advisories/a90146.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+toc: true
+affected_versions: v22.1.6 to v22.1.9
+advisory_date: 2022-12-19
```
Comment (taroface) [inline advisories/a90146.md:7]: Update to `2022-12-20`?

File: advisories/a90146.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+In CockroachDB [v22.1.6](../releases/v22.1.html#v22-1-6) to [v22.1.9](../releases/v22.1.html#v22-1-9), changefeeds using the [`initial_scan='only'`](../v22.1/create-changefeed#initial-scan) option can complete with a `successful` status even if not all messages have been emitted to the sink. This occurs consistently unless the table is very small.
+
+The same issue may occur if a changefeed with [`schema_change_policy='stop'`](../v22.1/create-changefeed#options) encounters a schema change. The changefeed may stop prior to all changes that occurred prior to the schema change being emitted.
```
Comment (taroface) [inline advisories/a90146.md:17]: "prior to all changes that occurred prior to" is a little hard for me to parse, due to the repetition of "prior". Would a word substitution and/or rephrase here help?

File: advisories/a90146.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Statement
+
+This is resolved in CockroachDB by [#90241](https://github.com/cockroachdb/cockroach/pull/90241), which ensures all events correctly flush out of all buffers prior to the changefeed exiting.
+
+The fix has been applied to maintenance releases of CockroachDB [v22.1.10](../releases/v22.1.html#v22-1-10) and later.
+
+This public issue is tracked by [#90146](https://github.com/cockroachdb/cockroach/pull/90146).
```
Comment (taroface) [inline advisories/a90146.md:25]: This link should technically be `https://github.com/cockroachdb/cockroach/issues/90146` even though the `pull` URL redirects correctly.

File: advisories/a90146.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+## Mitigation
+
+Users of CockroachDB [v22.1.6](../releases/v22.1.html#v22-1-6) to [v22.1.9](../releases/v22.1.html#v22-1-9) are encouraged to upgrade to [v22.1.10](../releases/v22.1.html#v22-1-10) or a later version. Prior and later versions are unaffected.
```
Comment (taroface) [inline advisories/a90146.md:29]: Not sure `Prior and later versions are unaffected.` is necessary. Plus it is not immediately clear what versions "prior and later" both refer to.

**Review by taroface — APPROVED** [review]
LGTM once a few nits are addressed.

File: advisories/a90146.md (hunk 5 above)
Comment (shermanCRL) [reply advisories/a90146.md:29]: Well, this was introduced in a particular patch release and resolved in a later patch release, so it’s a “window”. That said, the specific versions called out are clear enough, perhaps the “Prior and later...” is redundant.

**Pull request merged**